pkg/cmd/join/exec.go

```diff
@@ -193,7 +193,7 @@ func waitUntilRegistrationOperatorConditionIsTrue(f util.Factory, timeout int64)
 	operatorSpinner.Start()
 	defer operatorSpinner.Stop()
 
-	return waitUntilConditionIsTrue(
+	return helpers.WatchUntil(
```

Comment:
Followup: We should use "degraded" condition of klusterlet here instead of reading pod status.